app/scripts/controllers/buildConfig.js

```diff
@@ -74,6 +109,7 @@ angular.module('openshiftConsole')
                 };
               }
               $scope.buildConfig = buildConfig;
+              $scope.updatedBuildConfig = angular.copy(buildConfig);
```

Comment:
In addressing the other comments here, I noticed that changes to env vars made via `Edit YAML` are not reflected in the key-value-editor.  I assume this aforementioned bug is related to this same issue.